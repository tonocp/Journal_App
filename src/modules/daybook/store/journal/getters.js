// export const myGetter = ( state ) => {
// return state...
// };

export const getEntriesByTerm = (state) => {
  return state.entries;
};

export const getEntryById = (/* state */) => {};
