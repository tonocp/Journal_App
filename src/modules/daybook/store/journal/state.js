export default () => ({
  isLoading: true,
  entries: [
    {
      id: new Date().getTime(),
      date: new Date().toLocaleDateString(),
      text: "Sit excepteur id consectetur dolore amet dolore qui laboris eu nulla irure. Esse amet pariatur enim veniam proident ad proident. Adipisicing laborum laboris ea elit. Proident duis nulla officia esse culpa ullamco aute anim ut duis tempor.",
      picture: null,
    },
    {
      id: new Date().getTime() + 1000,
      date: new Date().toLocaleDateString(),
      text: "Culpa adipisicing amet enim anim cupidatat qui mollit nisi labore culpa tempor adipisicing ut est. Consectetur eu ad non adipisicing aliqua aliqua aliquip minim consequat officia sunt cupidatat veniam esse. Reprehenderit fugiat minim consequat adipisicing irure. Ea eiusmod duis incididunt non eiusmod laboris id qui deserunt velit pariatur velit. Commodo aliqua in id incididunt et enim ea dolor cillum sunt elit id pariatur eiusmod. Ipsum laborum eiusmod incididunt reprehenderit deserunt quis in irure dolor nisi laboris culpa ad esse.",
      picture: null,
    },
    {
      id: new Date().getTime() + 2000,
      date: new Date().toLocaleDateString(),
      text: "Anim amet labore labore aliqua commodo irure duis veniam ipsum Lorem est est sit. In culpa quis exercitation id proident. Ea irure occaecat labore consequat nulla irure commodo eu labore quis eu dolor Lorem sint. Deserunt deserunt ad pariatur laboris enim aliqua.",
      picture: null,
    },
  ],
});
