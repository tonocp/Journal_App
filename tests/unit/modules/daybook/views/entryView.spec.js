import { createStore } from "vuex";
import { shallowMount } from "@vue/test-utils";

import Swal from "sweetalert2";

import journal from "@/modules/daybook/store/journal";
import EntryView from "@/modules/daybook/views/EntryView.vue";
import { journalState } from "../../../mock-data/test-journal-state";

const createVuexStore = (initialState) =>
  createStore({
    modules: {
      journal: {
        ...journal,
        state: { ...initialState },
      },
    },
  });

jest.mock("sweetalert2", () => ({
  fire: jest.fn(),
  showLoading: jest.fn(),
  close: jest.fn(),
}));

describe("Pruebas en el EntryView", () => {
  const store = createVuexStore(journalState);
  store.dispatch = jest.fn();

  const mockRouter = {
    push: jest.fn(),
  };

  let wrapper;

  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = shallowMount(EntryView, {
      props: {
        id: "-Mr2XS8HZ41T8-mAxch7",
      },
      global: {
        mocks: {
          $router: mockRouter,
        },
        plugins: [store],
      },
    });
  });

  test("Debe de sacar al usuario porque el id no existe", () => {
    const wrapper = shallowMount(EntryView, {
      props: {
        id: "Este ID no existe en el STORE",
      },
      global: {
        mocks: {
          $router: mockRouter,
        },
        plugins: [store],
      },
    });

    expect(mockRouter.push).toHaveBeenCalledWith({ name: "no-entry" });
  });

  test("Debe de mostrar la entrada correctamente", () => {
    expect(wrapper.html()).toMatchSnapshot();
    expect(mockRouter.push).not.toHaveBeenCalled();
  });

  test("Debe de borrar la entrada y salir", (done) => {
    Swal.fire.mockReturnValueOnce(Promise.resolve({ isConfirmed: true }));

    wrapper.find(".btn-danger").trigger("click");

    expect(Swal.fire).toHaveBeenCalledWith({
      title: "¿Está seguro?",
      text: "Una vez borrado no podrá recuperar la entrada.",
      showDenyButton: true,
      confirmButtonText: "Borrar",
    });

    setTimeout(() => {
      expect(store.dispatch).toHaveBeenCalledWith("journal/deleteEntry", "-Mr2XS8HZ41T8-mAxch7");
      expect(mockRouter.push).toHaveBeenCalled();
      done();
    }, 1);
  });
});
