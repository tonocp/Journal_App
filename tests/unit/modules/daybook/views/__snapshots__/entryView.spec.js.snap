// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el EntryView Debe de mostrar la entrada correctamente 1`] = `
<div class="entry-title d-flex justify-content-between p-2">
  <div><span class="text-success fs-3 fw-bold">16</span><span class="mx-1 fs-3">Diciembre</span><span class="mx-2 fs-4 fw-light">2021, Jueves</span></div>
  <div><input type="file" accept="image/png, image/jpeg" style="display: none;"><button class="btn btn-danger mx-2"> Borrar <i class="fa fa-trash-all"></i></button><button class="btn btn-primary"> Subir foto <i class="fa fa-upload"></i></button></div>
</div>
<hr>
<div class="d-flex flex-column px-3 h-75"><textarea placeholder="¿Qué te cuentas hoy?"></textarea></div>
<!--v-if-->
<!--v-if-->
<fab-stub icon="fa-save"></fab-stub>
`;
